### PR TITLE
[Benchmark] Share data between SLA runs

### PR DESCRIPTION
**Pull request by DarkLight1337**
## Purpose

Further optimize the SLA script to share past run data coming from the same serve/bench combination. This is useful when tuning for multiple SLA targets (e.g. E2EL <= 200 ms, 500 ms, 1000 ms, ...) while keeping the serve/bench combination unchanged.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

---

> [!NOTE]
> Improves SLA sweeping by reusing prior results to speed up subsequent searches.
> 
> - Load existing `summary.json` files per `sla_variable` via `_iter_sla_val_paths` and pre-populate `SLAHistory` in `solve_sla`
> - Extract `_compute_margin` and use it for both loaded and new results
> - Tests: persist `summary.json` in mocked runs and add `test_solve_reuse_history` to verify cross-run history reuse; adjust existing tests to pass `base_path`
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e32b8fc89d7d94b36795d04e405f8d43f8acd581. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Reuses past SLA run data to accelerate subsequent searches and reduce redundant benchmarking.
> 
> - Add `_iter_sla_val_paths` to discover existing `summary.json` under `"<sla_variable>=<value>"` dirs and pre-fill `SLAHistory` in `solve_sla`
> - Extract `_compute_margin` to compute margins for both loaded and newly produced results
> - Adjust search initialization logic in `solve_sla` to handle pre-populated history and boundary cases
> - Test updates: persist `summary.json` in mocked `run_sla`, pass `base_path` via `tmp_path`, and add `test_solve_reuse_history` to verify cross-run reuse
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 29496561660ded853e90879c0c0f8d22e5ce6b0e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Speeds up SLA searches by reusing previously saved results for the same serve/bench combination.
> 
> - Add `_iter_sla_val_paths` to discover existing `summary.json` under `"<sla_variable>=<value>"` and preload `SLAHistory` in `solve_sla`
> - Extract `_compute_margin` and use it for both loaded and newly produced results; simplify margin computation in `solve_sla`
> - Refine initial value selection in `solve_sla` when history is pre-populated (handle bounds and already-tested values)
> - Tests: persist `summary.json` in mocked `run_sla`, pass `base_path` via `tmp_path`, and add `test_solve_reuse_history` to confirm history reuse
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1c775f8eccea9de2a8ebb05089fdf17d43f3fb61. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Improves SLA sweeping by reusing past run data for the same serve/bench combo to avoid redundant benchmarking and speed up searches.
> 
> - Load existing `summary.json` for each `"<sla_variable>=<value>"` via `_iter_sla_val_paths` and pre-seed `SLAHistory` in `solve_sla`
> - Extract `_compute_margin` and use it for both loaded and newly produced results; simplify margin computation in `solve_sla`
> - Refine initial value selection in `solve_sla` when history is pre-populated (handle bounds and skip already-tested values)
> - Tests: mock `run_sla` now writes `summary.json`; pass `base_path` in tests; add `test_solve_reuse_history` to verify cross-run reuse
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 107159df60b456cda649cd31464ef8a7cb76aa32. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Reuses previously saved SLA summaries to speed up subsequent searches for the same serve/bench combination.
> 
> - Load existing `summary.json` under `"<sla_variable>=<value>"` via `_iter_sla_val_paths` and pre-populate `SLAHistory` in `solve_sla`
> - Extract `_compute_margin` to compute margins for both loaded and newly produced results; simplify margin handling in `solve_sla`
> - Refine initial value selection and skip already-tested values during search
> - Tests: persist `summary.json` in mocked runs using `_get_sla_run_path`, pass `base_path` (`tmp_path`), and add `test_solve_reuse_history` to verify cross-run reuse
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit fb08da832b0caa46d600c7a54002fe1b4d7dad14. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Reuses prior SLA results to reduce redundant benchmarking and speed up searches for the same serve/bench combination.
> 
> - Add `_iter_sla_val_paths` to discover existing `"<sla_variable>=<value>"/summary.json` and preload `SLAHistory` in `solve_sla`
> - Extract `_compute_margin` and use it for both loaded summaries and newly produced results; simplify margin handling in `solve_sla`
> - Refine initial value selection when history is pre-populated (handle bounds and skip already-tested values)
> - Tests: mocked `run_sla` now writes `summary.json` via `_get_sla_run_path`; pass `base_path` in all tests; add `test_solve_reuse_history` to confirm cross-run reuse
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 25fecce4044f9064265d2192f5ca26f242c2b4e3. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: vllm/benchmarks/sweep/serve_sla.py [hunk 1]
```diff
@@ -170,6 +193,13 @@ def solve_sla(
     sla_data = list[dict[str, object]]()
     history = SLAHistory(min_value=sla_min_value, max_value=sla_max_value)
 
+    # Use results from previous runs
+    for past_sla_value, path in _iter_sla_val_paths(base_path, sla_variable):
+        with path.open("rb") as f:
+            past_iter_data = json.load(f)
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/benchmarks/sweep/serve_sla.py:203]: Fixed

File: vllm/benchmarks/sweep/serve_sla.py [hunk 2]
```diff
@@ -153,6 +159,23 @@ def get_min_failing(self) -> float:
...
+        k: sum(float(run_data[k]) for run_data in iter_data) / len(iter_data)  # type: ignore
+        for k in sla_comb
+    }
+
+    sla_margins = [
+        criterion.print_and_compute_margin(iter_data_mean, k)
+        for k, criterion in sla_comb.items()
+    ]
+
+    return max(sla_margins)
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/benchmarks/sweep/serve_sla.py:180]: Fixed

File: vllm/benchmarks/sweep/serve_sla.py [hunk 3]
```diff
@@ -170,6 +193,13 @@ def solve_sla(
     sla_data = list[dict[str, object]]()
     history = SLAHistory(min_value=sla_min_value, max_value=sla_max_value)
 
+    # Use results from previous runs
+    for past_sla_value, path in _iter_sla_val_paths(base_path, sla_variable):
+        with path.open("rb") as f:
+            past_iter_data = json.load(f)
+
+        history[past_sla_value] = _compute_margin(sla_comb, past_iter_data)
+
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/benchmarks/sweep/serve_sla.py:206]: Fixed

**Review by Isotr0py — APPROVED** [review]
